### PR TITLE
Made SyncManager extend ComponentManager

### PR DESCRIPTION
**Pull request by damithc**

### CODE REVIEW

**Comment by damithc** [comment]
@m133225 for your review and merging

File: src/main/java/address/sync/SyncManager.java [hunk 1]
```diff
@@ -28,18 +29,16 @@
     private RemoteManager remoteManager;
 
     public SyncManager() {
-        activeAddressBook = Optional.empty();
-        scheduler = Executors.newScheduledThreadPool(1);
-        requestExecutor = Executors.newCachedThreadPool();
-        EventManager.getInstance().registerHandler(this);
```
Comment (m133225) [inline src/main/java/address/sync/SyncManager.java]: I think this shouldn't be removed? Needed for `SyncManager` to handle certain events.

Comment (HansNewbie) [inline src/main/java/address/sync/SyncManager.java]: It's in `super()` call of `ComponentManager`.

**Comment by damithc** [comment]
@m133225 go ahead and merge if ok.

**Pull request merged**